src/convertToObject.js

```diff
@@ -5,8 +5,23 @@
  *
  * @return {object}
  */
+const SEMICOLON = ';';
+const COLON = ':';
+
 function convertToObject(sourceString) {
-  // write your code here
+  const styles = sourceString.split(SEMICOLON);
+
+  const stylesObject = styles.reduce((temp, style) => {
+    const separateStyle = style.split(COLON).map((part) => part.trim());
```

Comment:
[CODE KNOWLEDGE] - There's no need to make a copy before each operation. In this case, the 'map' method does not mutate the original array, so it can be used directly without the need for copying.